types/react-test-renderer/react-test-renderer-tests.ts

```diff
@@ -66,3 +66,12 @@ const shallowRenderer = createRenderer();
 shallowRenderer.render(component);
 shallowRenderer.getRenderOutput();
 shallowRenderer.getMountedInstance();
+
+// Only synchronous, void callbacks are acceptable for act()
+act(() => {});
+// $ExpectError
+act(async () => {});
+// $ExpectError
+act(() => null);
+// $ExpectError
+Promise.resolve(act(() => {}));
```

Comment:
This errors for 2 reasons.

First, the `| {}` causes an error because the compiler (correctly) thinks there might not be a `then` it can access.

Second, even if the `| {}` is somehow erased, the `DebugPromiseLike` itself fails to be usable as a `PromiseLike` because of the type of the _second_ argument.